torch/testing/_internal/optests/generate_tests.py

```diff
@@ -0,0 +1,318 @@
+
+    def __exit__(self, *args, **kwargs):
+        try:
+            self.maybe_raise_errors_on_exit()
+        finally:
+            result = super().__exit__(*args, **kwargs)
+        return result
+
+    def run_test_util(self, op, args, kwargs):
+        try:
+            self.test_util(op, args, kwargs)
+        except torch._subclasses.fake_tensor.UnsupportedFakeTensorException:
+            # We might get here if the input is already a FakeTensor
+            # or if we're in a torch.compile block. Just ignore these
+            # since we can't handle them and reporting them as failures
+            # is too noisy.
+            pass
+
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs if kwargs else {}
+
+        # Only intercept calls to operators
+        if not isinstance(func, (torch._ops.OpOverloadPacket, torch._ops.OpOverload)):
+            return func(*args, **kwargs)
+        # Pre-existing code may not use the .default overload. If we see an
+        # OpOverloadPacket and we cannot resolve the overload, then we just throw
+        # and ask the user to clarify. Otherwise, we attempt to resolve the overload.
+        if isinstance(func, torch._ops.OpOverloadPacket):
+            func = resolve_unique_overload_or_throw(func)
+        qualname = func.name()
+        ns = qualname.split("::")[0]
+        if ns not in self.namespaces:
+            return func(*args, **kwargs)
+
+        args_c, kwargs_c = deepcopy_tensors(args, kwargs)
+        # Only call test_util(op, *args, **kwargs) if this succeeds.
+        result = func(*args, **kwargs)
+
+        option = retrieve(self.failures_dict, qualname, self.test_name)
+        if option == "success" or option == "xfail":
+            # Surpress all errors during execution. Raise them during __exit__.
+            try:
+                if qualname not in self.seen_ops_to_errors:
+                    self.seen_ops_to_errors[qualname] = []
+                self.run_test_util(func, args_c, kwargs_c)
+            except Exception as ex:
+                self.seen_ops_to_errors[qualname].append(ex)
+        elif option == "skip":
+            pass
+        return result
+
+class OpCheckError(Exception):
+    pass
+
+
+def resolve_unique_overload_or_throw(op: torch._ops.OpOverloadPacket):
+    all_schemas = torch._C._jit_get_schemas_for_operator(op._qualified_op_name)
+    if len(all_schemas) != 1:
+        raise RuntimeError(
+            f"opcheck can only test operators without overloads. "
+            f"Got the following overloads for {op._qualified_op_name}: "
+            f"{[schema.overload_name for schema in all_schemas]}"
+        )
+
+    overload_name = all_schemas[0].overload_name
+    if overload_name == "":
+        return op.default
+    return getattr(op, overload_name)
+
+
+def deepcopy_tensors(args, kwargs):
+    def deepcopy_tensor(x):
+        return x.detach().clone().requires_grad_(x.requires_grad)
```

Comment:
Nyeh, this is not as good as some of the utils in dynamo. In particular this won't preserve striding. Granted, most FB custom ops don't deal with striding, but we should ourselves to a higher standard.